nixos/platform/systemd.nix

```diff
@@ -25,18 +25,22 @@ in
           "service"
           "admins"
         ];
-        acls =
-          lib.concatMapStrings
-            (group: "-m g:${group}:rX -m d:g:${group}:rX ")
-            journalReadGroups;
 
-      in ''
+      mkSetfaclCmd = group: ''
+        if [ $(getent group ${group}) ]; then
+          ${pkgs.acl}/bin/setfacl -R -m g:${group}:rX -m d:g:${group}:rX /var/log/journal
+        else
+          echo "Warning: expected group '${group}' not found, skipping ACL."
+        fi
+      '';
+
+      in lib.stringAfter [ "groups" ] (''
```

Comment:
I really like the alternative notation of activation scripts as an attrset with

```nix
{
  text = ''the script'';
  deps = [ "some" "other" "activation" "scripts" "like" "groups" ];
}
```

That makes it more clear what happens there IMHO. What do you think about using this for new or modified activation scripts from now on?